### ISSUE TITLE
AvoidFor cop misses many violations

### ISSUE THREAD

**Issue by jonas054** [rubocop/rubocop#159]
For example in:

``` ruby
def func
  for i in (0..2)
    puts i
  end
end
```

no offences are detected.

I'm working on a solution.
